ITE/10/README.adoc

```diff
@@ -0,0 +1,288 @@
+
+| Status
+| Draft
+
+| Type
+| Standards
+
+| Created
+| 2023-01-07
+
+|===
+
+[[abstract]]
+== Abstract
+
+A previous in-toto Enhancement, ITE-6, presented the in-toto Attestation
+Framework. This framework introduced the idea of context-specific attestations
+with their own definitions. This ITE builds on ITE-6 and proposes several
+modifications to in-toto layouts that allow for defining and verifying policies
+over ITE-6 attestations.
+
+[[specification]]
+== Specification
+
+The in-toto specification describes the following schema for layouts.
+
+```json
+{
+    "_type": "layout",
+    "expires": "<EXPIRES>",
+    "readme": "<README>",
+    "keys": {
+        "<KEYID>": "<PUBKEY_OBJECT>"
+    },
+    "steps": ["<STEP>", "..."],
+    "inspect": ["<INSPECTION>", "..."]
+}
+```
+
+An in-toto layout provides two semantics for software supply chain owners to
+define policies: steps and inspections. Steps are used to reason about and
+express constraints regarding operations _already_ performed as part of the
+supply chain. The verification is focused on ensuring authorized functionaries
+performed each step and validating the integrity of artifacts as they move from
+one step to the next. Inspections are executed _during_ verification to perform
+any other checks that are defined as external scripts.
+
+TODO: this pass does not revise artifact rules for steps.
+
+This ITE primarily updates the schema and capabilities for inspections. As it
+stands, the inspections semantic can already be used to apply arbitrary
+constraints to data captured in in-toto links. Similarly, they can apply
+constraints to attestations as well. We update in-toto inspections with more
+capabilities so that they do not necessarily have to be executed as separate
+scripts. This is achieved by providing multiple options for the domain specific
+language used to express constraints. Currently, inspections only support using
+in-toto's artifact rules. With this ITE, this will remain an option, but
+inspections can also be written to not execute separate shell scripts and use
+artifact rules, and instead express data constraints using DSLs like Rego or
+Cue. The current inspection schema is as follows:
+
+```json
+{
+    "_type": "inspection",
+    "name": "<NAME>",
+    "expected_materials": [
+        [ "<ARTIFACT_RULE>" ],
+        "..."
+    ],
+    "expected_products": [
+        [ "<ARTIFACT_RULE>" ],
+        "..."
+    ],
+    "run": "<COMMAND>"
+}
+```
+
+The above schema can also be expressed using the following generalization that
+allows for multiple DSLs. Note that several fields are optional.
+
+
+```json
+{
+    "_type": "inspection",
+    "name": "<NAME>",
+    "functionaries": ["<KEYID>", "..."],
+    "threshold": "<THRESHOLD>",
+    "subject": ["<SUBJECT>", "..."],
+    "policyLanguage": "<POLICY_LANGUAGE>",
+    "policy": "<POLICY>",
+    "run": "<COMMAND>"
+}
+```
+
+Currently supported inspections _always_ execute a command or script and
```

Comment:
Yes, and that's the current state of an inspection give or take. Currently, inspections are invoked using the `run` field in the layout. This may be a direct command such as a decompress as seen in in-toto/demo or the execution of a script shipped alongside the layout.